### PR TITLE
FM2-418: Support for type-level $everything operation on Patients

### PR DESCRIPTION
**Pull request by Medhavi-16**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-418

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by Medhavi-16** [comment]
@ibacher @varung-31 PTAL.

File: api/src/main/java/org/openmrs/module/fhir2/api/FhirPatientService.java [hunk 1]
```diff
@@ -44,4 +44,6 @@ IBundleProvider searchForPatients(StringAndListParam name, StringAndListParam gi
 	        HashSet<Include> revIncludes);
 	
 	IBundleProvider getPatientEverything(TokenParam identifier);
+	
+	IBundleProvider getPatientEverythingType();
```
Comment (varung-31) [inline api/src/main/java/org/openmrs/module/fhir2/api/FhirPatientService.java:48]: We could have overloaded methods.
```suggestion
	IBundleProvider getPatientEverything();
```

File: api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirPatientServiceImpl.java [hunk 2]
```diff
@@ -115,6 +115,21 @@ public IBundleProvider getPatientEverything(TokenParam patientId) {
...
+	@Transactional(readOnly = true)
+	public IBundleProvider getPatientEverythingType() {
+		SearchParameterMap theParams = new SearchParameterMap().addParameter(FhirConstants.EVERYTHING_SEARCH_HANDLER, "");
+		
+		getEverythingReverseInclude(theParams);
+		return searchQuery.getQueryResults(theParams, dao, translator, searchQueryInclude);
+		
+	}
+	
+	private void getEverythingReverseInclude(SearchParameterMap theParams) {
```
Comment (varung-31) [inline api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirPatientServiceImpl.java:132]: `getEverythingReverseInclude` feels like the method returns us something.
A better name should be `populateEverythingOperationParams`

File: omod/src/test/java/org/openmrs/module/fhir2/providers/r4/PatientFhirResourceProviderWebTest.java [hunk 3]
```diff
@@ -805,4 +812,23 @@ private void verifyEverythingOperation(String uri) throws Exception {
...
+		Patient patient = new Patient();
+		patient.setId(PATIENT_UUID);
+		
+		when(patientService.getPatientEverythingType())
+		        .thenReturn(new MockIBundleProvider<>(Collections.singletonList(patient), 10, 1));
+		
+		MockHttpServletResponse response = get(uri).accept(FhirMediaTypes.JSON).go();
+		
+		MatcherAssert.assertThat(response, isOk());
+		MatcherAssert.assertThat(response.getContentType(), equalTo(FhirMediaTypes.JSON.toString()));
```
Comment (varung-31) [inline omod/src/test/java/org/openmrs/module/fhir2/providers/r4/PatientFhirResourceProviderWebTest.java:825]:
```suggestion
		assertThat(response, isOk());
		assertThat(response.getContentType(), equalTo(FhirMediaTypes.JSON.toString()));
```

**Review by varung-31 — CHANGES_REQUESTED** [review]
Great work @Medhavi-16! Some minor comments and it should be good.

File: omod/src/test/java/org/openmrs/module/fhir2/providers/r4/PatientFhirResourceProviderWebTest.java (hunk 3 above)
Comment (Medhavi-16) [reply omod/src/test/java/org/openmrs/module/fhir2/providers/r4/PatientFhirResourceProviderWebTest.java]: @ibacher @varung-31 PTAL

**Comment by varung-31** [comment]
LGTM! @ibacher Do you have anything to add?

File: omod/src/test/java/org/openmrs/module/fhir2/providers/r4/PatientFhirResourceProviderWebTest.java (hunk 3 above)
Comment (ibacher) [reply omod/src/test/java/org/openmrs/module/fhir2/providers/r4/PatientFhirResourceProviderWebTest.java:825]: @Medhavi-16 Has this been addressed?

**Comment by ibacher** [comment]
@Medhavi-16 I had to do some Git gymnastic to create a point release for this module. Would you be able to fix the resulting conflicts? Thanks! Otherwise, this LGTM.

**Comment by Medhavi-16** [comment]
> @Medhavi-16 I had to do some Git gymnastic to create a point release for this module. Would you be able to fix the resulting conflicts? Thanks! Otherwise, this LGTM.

Yes, @ibacher I'll resolve the conflicts.

**Comment by Medhavi-16** [comment]
@ibacher @varung-31 PTAL.

**Review by ibacher — APPROVED** [review]

**Comment by ibacher** [comment]
Thanks @Medhavi-16!

**Pull request merged**